docs/DeveloperGuide.md

```diff
@@ -1,38 +1,219 @@
+- `help` : Shows the help page.  
+- `bye` : Exits the program.  
+
+```
+____________________________________________________________
+____    __    ____  __   __       _______
+\   \  /  \  /   / |  | |  |     |       \
+ \   \/    \/   /  |  | |  |     |  .--.  |
+  \            /   |  | |  |     |  |  |  |
+   \    /\    /    |  | |  `----.|  '--'  |
+    \__/  \__/     |__| |_______||_______/
+____    __    ____  ___   .___________.  ______  __    __
+\   \  /  \  /   / /   \  |           | /      ||  |  |  |
+ \   \/    \/   / /  ^  \ `---|  |----`|  ,----'|  |__|  |
+  \            / /  /_\  \    |  |     |  |     |   __   |
+   \    /\    / /  _____  \   |  |     |  `----.|  |  |  |
+    \__/  \__/ /__/     \__\  |__|      \______||__|  |__|
+____________________________________________________________
+Hello there! Welcome to WildWatch!
+
+Checking if "WildWatch.txt" already exists...
+File does not exist.
+Creating new file...
+File created successfully.
+What would you like to do?
+____________________________________________________________
+>>> 
+```
+
+[⬆ Back to top](https://ay2324s1-cs2113t-w11-2.github.io/tp/DeveloperGuide.html#table-of-contents)
+
+--------------------------------------------------------------------------------------------------------------------------------------
+
+## Design & Implementation
+
+### System Architecture
+
+![System Architecture](https://github.com/AY2324S1-CS2113T-W11-2/tp/assets/69474977/90309a3c-f784-4ffb-8eef-65735c05ec52)
 
-## Acknowledgements
+The **_Architecture Diagram_** above shows a high-level overview of the architectural design of WildWatch.  
+Actual detailed implementation may differ from the architectural diagram.  
+(e.g., `Parser` in the diagram encompasses the `Input`, `Error`, and `Parser` packages).  
+  
+WildWatch is comprised of 7 major components.  
+- `Main`: Entry point of the program.
+- `UI`: A class that receives input from, and prints output to the user.
+- `Parser`: A class that processes and interpretes the input command of the user.
+- `Command`: A set of classes that does specific task according to the command of the user.
+- `EntryList`: A class that stores all the wildlife data, while the program is running.
+- `Entry`: A class that stores individual wildlife data, while the program is running.
+- `FileHandler`: A class that saves and retrieves all the wildlife data from the local storage
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+![Architecture Sequence Diagram](https://github.com/AY2324S1-CS2113T-W11-2/tp/assets/69474977/6bea5165-aa78-4b3e-baf0-2e9ced109161)  
 
-## Design & implementation
+The **_Generic Sequence Diagram_** above shows a shows how the components in the architecture interact with each other for a generic command input in WildWatch.  
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/38713736/d899421f-a7b5-43c6-a1cd-d294bc020bb2)
It might be better to seperate error handling using `checkError()` method from the construction of a Parser object